module/module_test.go

```diff
@@ -493,192 +463,124 @@ func TestModularResourceWithoutReconfigure(t *testing.T) {
+		test.That(t, deps, test.ShouldResemble, []string{"rdk:component:motor/motor1"})
 
-	myRobot, err := robotimpl.RobotFromConfig(ctx, cfg, logger)
-	test.That(t, err, test.ShouldBeNil)
+		_, err = m.AddResource(ctx, &pb.AddResourceRequest{
+			Config: mockConf, Dependencies: deps,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	parentAddr, err := myRobot.ModuleAddress()
-	test.That(t, err, test.ShouldBeNil)
+		test.That(t, createDeps1, test.ShouldResemble, deps)
+		test.That(t, createConf1.Attributes, test.ShouldResemble, mockConf.Attributes)
+		test.That(t, createConf1.ConvertedAttributes, test.ShouldImplement, &MockConfig{})
+	})
 
-	addr := filepath.ToSlash(filepath.Join(filepath.Dir(parentAddr), "mod.sock"))
-	m, err := module.NewModule(ctx, addr, logger)
-	test.That(t, err, test.ShouldBeNil)
+	//nolint:dupl
+	t.Run("non-reconfigurable recreation", func(t *testing.T) {
+		mockAttrs, err := protoutils.StructToStructPb(MockConfig{
+			Motors: []string{motor.Named("motor2").String()},
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	model := resource.NewModel("mock", "demo", "mocksummation")
-	var mock *mockSummationWithReconfigure
+		mockConf.Attributes = mockAttrs
 
-	// Register the mock summation service
-	registry.RegisterService(summationapi.Subtype, model, registry.Service{
-		Constructor: func(ctx context.Context, deps registry.Dependencies, cfg config.Service, logger *zap.SugaredLogger) (interface{}, error) {
-			conf, ok := cfg.ConvertedAttributes.(*MockSummationConfig)
-			res := &mockSummationWithReconfigure{mockSummation{deps: deps, config: conf}, 0}
-			if !ok {
-				return nil, rdkutils.NewUnexpectedTypeError(conf, cfg.ConvertedAttributes)
-			}
+		validateResp, err := m.ValidateConfig(ctx, &pb.ValidateConfigRequest{
+			Config: mockConf,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-			mock = res
-			return res, nil
-		},
-	})
-	defer func() {
-		// Deregister the mock summation service
-		registry.DeregisterService(summationapi.Subtype, model)
-	}()
+		deps := validateResp.Dependencies
+		test.That(t, deps, test.ShouldResemble, []string{"rdk:component:motor/motor2"})
 
-	// Register the mock summation service attribute map converter
-	config.RegisterServiceAttributeMapConverter(
-		summationapi.Subtype,
-		model,
-		func(attributes config.AttributeMap) (interface{}, error) {
-			var conf MockSummationConfig
-			return config.TransformAttributeMapToStruct(&conf, attributes)
-		},
-		&MockSummationConfig{})
+		_, err = m.ReconfigureResource(ctx, &pb.ReconfigureResourceRequest{
+			Config: mockConf, Dependencies: deps,
+		})
+		test.That(t, err, test.ShouldBeNil)
 
-	sumAttrs, err := protoutils.StructToStructPb(MockSummationConfig{
-		Motors: []string{motor.Named("motor1").String()},
+		test.That(t, createDeps1, test.ShouldResemble, deps)
+		test.That(t, createConf1.Attributes, test.ShouldResemble, mockConf.Attributes)
+		test.That(t, createConf1.ConvertedAttributes, test.ShouldImplement, &MockConfig{})
```

Comment:
Shouldn't this test the actual values within ConvertedAttributes? Not just that it implements MockConfig{}? Zeroed out MockConfig{} would pass this test but not be valid behavior.